packages/mini-browser/src/browser/mini-browser-open-handler.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { ApplicationShell } from '@theia/core/lib/browser/shell';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import { LabelProvider } from '@theia/core/lib/browser/label-provider';
+import { WidgetOpenHandler, WidgetOpenerOptions } from '@theia/core/lib/browser/widget-open-handler';
+import { FileLocationMapper } from './location-mapper-service';
+import { MiniBrowser, MiniBrowserProps } from './mini-browser';
+
+/**
+ * Further options for opening a new `Mini Browser` widget.
+ */
+export interface MiniBrowserOpenerOptions extends WidgetOpenerOptions, MiniBrowserProps {
+
+}
+
+@injectable()
+export class MiniBrowserOpenHandler extends WidgetOpenHandler<MiniBrowser> {
+
+    readonly id = 'mini-browser-open-handler';
+    readonly label = 'Mini Browser';
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
+    async canHandle(uri: URI): Promise<number> {
+        const url = await FileLocationMapper.toURL(uri, 'mini-browser-check');
+        const response = await fetch(url);
+        return response.status === 200 ? 500 : 0;
```

Comment:
The editor should be used by default.

Reply:
I have to reduce the priority so that the default code editor opens on double click.